assets/js/modules/analytics-4/utils/account.js

```diff
@@ -67,3 +67,48 @@ export function getAccountDefaults(
 		[ ENHANCED_MEASUREMENT_ENABLED ]: true,
 	};
 }
+
+/**
+ * Appends an account ID to an account object.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} account Account object.
+ * @param {string} idKey   Key of the account ID.
+ * @return {Object} Account object with an appended ID.
+ */
+export const appendAccountID = ( account, idKey = 'account' ) => {
+	const obj = { ...account };
+
+	const matches = account[ idKey ].match( /accounts\/([^/]+)/ );
+	if ( matches ) {
+		obj._id = matches[ 1 ];
+	}
+
+	return obj;
+};
+
+/**
+ * Appends property and account IDs to a property object.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} property Property object.
+ * @param {string} idKey    Key of the property ID.
+ * @return {Object} Property object with appended IDs.
+ */
+export const appendPropertyAndAccountIds = ( property, idKey = 'property' ) => {
+	const obj = { ...property };
```

Comment:
Rather than cloning and then mutating, I'd suggest using variables for the properties or just a partial object and then returning a new object at the end with spread, 
e.g. ` return { ...property, _id, _accountID }`

This also avoids the need for an ambiguous `obj` variable (which is good to avoid) 👍